packages/esm-patient-chart-app/src/deceased/deceased.resource.ts

```diff
@@ -0,0 +1,137 @@
+import { useMemo } from 'react';
+
+import { openmrsFetch, toOmrsIsoString, toDateObjectStrict } from '@openmrs/esm-framework';
+import useSWR from 'swr';
+import dayjs from 'dayjs';
+
+interface PersonFetchResponse {
+  person: { uuid: string; dead: boolean; deathDate: Date; type: string };
+}
+
+interface CauseOfDeathFetchResponse {
+  uuid: string;
+  value: string;
+}
+
+interface ConcentAnswersResponse {
+  answers: Array<{
+    uuid: string;
+    name: string;
+    display: string;
+  }>;
+}
+
+export function usePatientDeceased(patientUuid: string) {
+  const person = usePersonfromPatient(patientUuid);
+  const causeOfDeath = useCauseOfDeathConcept();
+  const conceptAnswers = useConceptAnswers(causeOfDeath.value);
+  const result = useMemo(() => {
+    return {
+      conceptAnswers: conceptAnswers.conceptAnswers,
+      deathDate: person.personObject?.deathDate,
+      personUuid: person.personObject?.uuid,
+      isDead: person.personObject?.dead,
+      isLoading: person.isPersonLoading && causeOfDeath.isCauseOfDeathLoading && causeOfDeath.isCauseOfDeathLoading,
+      isValidating:
+        person.isPersonValidating && conceptAnswers.isConceptAnswerValidating && causeOfDeath.isCauseOfDeathValidating,
+      refetchPatient: person.mutate,
+    };
+  }, [
+    causeOfDeath.isCauseOfDeathLoading,
+    causeOfDeath.isCauseOfDeathValidating,
+    conceptAnswers,
+    person.isPersonLoading,
+    person.isPersonValidating,
+    person.personObject?.dead,
+    person.personObject?.deathDate,
+    person.personObject?.uuid,
+    person.mutate,
+  ]);
+  return result;
+}
+
+const changePatientDeathStatus = (
+  deathDate: Date,
+  dead: boolean,
+  personUuid: string,
+  causeOfDeath: string,
+  abortController: AbortController,
+) => {
+  return openmrsFetch(`/ws/rest/v1/person/${personUuid}`, {
+    headers: {
+      'Content-type': 'application/json',
+    },
+    method: 'POST',
+    body: { deathDate, dead, causeOfDeath },
+    signal: abortController.signal,
+  });
+};
+
+export function markPatientDeceased(
+  deceasedDate: Date | null,
+  personUuid: string,
+  selectedCauseOfDeathValue: string | null,
+  abortController: AbortController,
+) {
+  const payload = {
+    deathDate: toDateObjectStrict(
```

Comment:
We are using 2 Date objects, omrsISOString and dayjs, which is not efficient by the looks. What do you trying to do here?

Reply:
So it's true this looks off but it looks like the `toDateObjectStrict` requires a date in that format I have looked in many places that is the way we have been sending the date. check this https://github.com/openmrs/openmrs-esm-patient-chart/blob/39a9dd30fc9ced3d06df75e8f0f8e0c9786853c4/packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx#L94